control/controlhttp/client.go

```diff
@@ -528,9 +541,16 @@ func (a *Dialer) tryURLUpgrade(ctx context.Context, u *url.URL, optAddr netip.Ad
 
 	tr := http.DefaultTransport.(*http.Transport).Clone()
 	defer tr.CloseIdleConnections()
-	tr.Proxy = a.getProxyFunc()
-	tshttpproxy.SetTransportGetProxyConnectHeader(tr)
-	tr.DialContext = dnscache.Dialer(dialer, dns)
+	if optACEHost != "" {
+		// If using ACE, we don't want to use any HTTP proxy.
+		// ACE is already a tunnel+proxy.
+		tr.Proxy = nil
+		tr.DialContext = dialer
```

Comment:
Non-blocking: I don’t think that it hurts to go through a proxy hosted by the end-user’s network, so I wouldn’t bother with this special case.

Reply:
We can add it later. I'd like to start without it, especially until we have good integration tests.

Reply:
done